### PR TITLE
Upgrade sbt, Play, and various dependencies

### PR DESCRIPTION
**Pull request by rtyley**
Upgrading sbt addresses the Log4J issue, there are various security updates as well going from Play 2.6 to Play 2.8.

### CODE REVIEW

File: app/model/Instance.scala [hunk 1]
```diff
@@ -105,7 +105,7 @@ case class StandardWebApp(versionUrl: String) extends AppSpecifics {
   )
 
   val versionExtractor = { r: WSResponse =>
-    val values = r.body.lines.map(_.split(':').map(_.trim)).collect { case Array(k, v) => k -> v }.toMap
+    val values = r.body.linesIterator.map(_.split(':').map(_.trim)).collect { case Array(k, v) => k -> v }.toMap
```
Comment (rtyley) [inline app/model/Instance.scala:108]: See https://github.com/scala/bug/issues/11125

File: app/lib/ScheduledAgent.scala [hunk 2]
```diff
@@ -17,16 +18,16 @@ object ScheduledAgent {
 
 class ScheduledAgent[T](initialDelay: FiniteDuration, frequency: FiniteDuration, initialValue: T, block: => Future[T], system: ActorSystem)(implicit ec: ExecutionContext) {
 
-  val agent = Agent[T](initialValue)
+  val agent = new AtomicReference[T](initialValue)
```
Comment (rtyley) [inline app/lib/ScheduledAgent.scala:21]: Akka Agents are [deprecated](https://doc.akka.io/docs/akka/2.5.32/agents.html#introduction), [`AtomicReference`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReference.html) is a simple substitute.

File: app/lib/Config.scala [hunk 3]
```diff
@@ -52,15 +52,15 @@ object Config {
   def configuration =
     Configuration(fileConfig(localPropsFile).withFallback(ConfigFactory.load()))
 
-  lazy val proxyHost = configuration.getString("proxyHost")
-  lazy val proxyPort = configuration.getInt("proxyPort")
+  lazy val proxyHost = configuration.getOptional[String]("proxyHost")
+  lazy val proxyPort = configuration.getOptional[Int]("proxyPort")
 
-  lazy val managementPort = configuration.getInt("managementPort")
+  lazy val managementPort = configuration.getOptional[Int]("managementPort")
```
Comment (rtyley) [inline app/lib/Config.scala:58]: Changes in Play's configuration library [introduced with Play 2.6](https://www.playframework.com/documentation/2.8.x/Migration26#Scala-Configuration-API).

File: build.sbt [hunk 4]
```diff
@@ -8,37 +8,40 @@ enablePlugins(PlayScala, SbtWeb, RiffRaffArtifact, BuildInfoPlugin, JDebPackagin
...
 resolvers += "scalaz-bintray" at "https://dl.bintray.com/scalaz/releases"
 
-scalaVersion := "2.12.10"
+scalaVersion := "2.12.15"
 scalacOptions ++= List("-feature", "-deprecation")
 
 val jacksonVersion = "2.12.3"
 
+// Until all dependencies are on scala-java8-compat v1.x, this avoids unnecessary fatal eviction errors
+ThisBuild / libraryDependencySchemes += "org.scala-lang.modules" %% "scala-java8-compat" % VersionScheme.Always
```
Comment (rtyley) [inline build.sbt:17]: See eg https://github.com/akka/akka/pull/30375, https://github.com/scanamo/scanamo/pull/1285 for explanation about this issue.

File: build.sbt [hunk 5]
```diff
@@ -8,37 +8,40 @@ enablePlugins(PlayScala, SbtWeb, RiffRaffArtifact, BuildInfoPlugin, JDebPackagin
...
+
 libraryDependencies ++= Seq(
-  "com.amazonaws" % "aws-java-sdk" % "1.11.890",
```
Comment (rtyley) [inline build.sbt:17]: This dependency pulls in the whole of the AWS SDK (which has [literally hundreds of different jars in it](https://search.maven.org/artifact/com.amazonaws/aws-java-sdk/1.12.129/jar)) - it's better to specify just the dependencies you need (`"dynamodb", "ec2", "elasticloadbalancing", "s3", "autoscaling", "cloudwatch", "sqs"` in this case, added below). This gives a dramatic reduction in overall deployment artifact size - from 249MB to 89MB in this case:

```
$aws s3 ls --profile deployTools --human-readable "s3://riffraff-artifact/status-app/147/status-app/status-app_1.0_all.deb"
2021-12-14 17:51:32  249.2 MiB status-app_1.0_all.deb
$ aws s3 ls --profile deployTools --human-readable "s3://riffraff-artifact/status-app/149/status-app/status-app_1.0_all.deb"
2021-12-14 17:53:09   89.0 MiB status-app_1.0_all.deb
```

File: build.sbt [hunk 6]
```diff
@@ -8,37 +8,40 @@ enablePlugins(PlayScala, SbtWeb, RiffRaffArtifact, BuildInfoPlugin, JDebPackagin
...
-  "com.typesafe.akka" %% "akka-agent" % "2.5.16",
   specs2 % Test,
   ehcache,
   ws,
-  "com.gu.play-googleauth" %% "play-v26" % "1.0.7",
-  "com.google.guava" % "guava" % "25.0-jre", //-- added explicitly - snyk report avoid logback vulnerability
```
Comment (rtyley) [inline build.sbt:23]: Not necessary to explicitly specify this dependency any more, at least not to get this version - Guava `30.1.1-jre` is pulled in by other dependencies.

File: conf/application.conf [hunk 7]
```diff
@@ -5,7 +5,7 @@
 # ~~~~~
 # The secret key is used to secure cryptographics functions.
 # If you deploy your application to several instances be sure to use the same key!
-application.secret="QjNS;iFjE;T6rLNZOiB7XUVC]Lo/Q=iDAYhX9W4>0KBad/qTR`trtL3f[geJJIp0"
+play.http.secret.key="QjNS;iFjE;T6rLNZOiB7XUVC]Lo/Q=iDAYhX9W4>0KBad/qTR`trtL3f[geJJIp0"
```
Comment (rtyley) [inline conf/application.conf:8]: This configuration key has changed name over time, [`play.http.secret.key` is the current name](https://www.playframework.com/documentation/2.8.x/ApplicationSecret).

Note that introducing https://github.com/guardian/play-secret-rotation would probably be a better idea that having this committed to the repository...

**Review by lmath — APPROVED** [review]
Nice one! ✨

**Pull request merged**